### PR TITLE
botocore - support tag for AWS RequestId

### PR DESCRIPTION
**Pull request by someboredkiddo**
This library has been an amazing tool for us debugging AWS services calls, but the first thing Amazon Support asks for during troubleshooting is the AWS requestid associated with the API call, which can be tricky to find. This PR will add that as a tag so I can remove some custom logging our apps are currently doing and just look at the DataDog APM web UI.

Opening this PR for general approach feedback. If it looks good I'm happy to update the rest of the `tests/contrib/botocore/test.py` test cases to assert the expected requestid matches motos' fixture.

(Also happy to open a github Issue but figured it was easier to see the code than describe the code.)

### CODE REVIEW

File: ddtrace/contrib/botocore/patch.py [hunk 1]
```diff
@@ -69,8 +69,13 @@ def patched_api_call(original_func, instance, args, kwargs):
 
         result = original_func(*args, **kwargs)
 
-        span.set_tag(http.STATUS_CODE, result['ResponseMetadata']['HTTPStatusCode'])
-        span.set_tag('retry_attempts', result['ResponseMetadata']['RetryAttempts'])
+        response_meta = result['ResponseMetadata']
+        span.set_tag(http.STATUS_CODE, response_meta['HTTPStatusCode'])
+        span.set_tag('retry_attempts', response_meta['RetryAttempts'])
+
+        request_id = response_meta.get('RequestId')
```
Comment (someboredkiddo) [inline ddtrace/contrib/botocore/patch.py]: not sure guarding against an unset RequestId is really necessary, but saw [aiobotocore](https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/contrib/aiobotocore/patch.py#L116) was cautious so figured I would be as well

Comment (brettlangdon) [reply ddtrace/contrib/botocore/patch.py]: doesn't hurt

**Comment by brettlangdon** [comment]
CircleCI is acting up a bit today, but seems like the boto tests are fine, so I've got it from here

**Comment by someboredkiddo** [comment]
thanks 👍  - let me know if you'd like to me to add another commit updating the rest of the `tests/contrib/botocore/test.py` test cases to assert the expected requestid matches motos' fixture.


For my own curiosity ...
This change seemed beneficial for all users and worth adding back to the official package ... But if I had a more custom/esoteric tag (something only meaningful to my org) and wanted to add it to a contrib's patch method but not back to DD's package, is there a way for an application to do that only through ddtrace library methods?

Only thing I could think of would be if my app monkeypatched the `contrib.<module>.patch.patch()` call with the updated code, but maybe I missed something?

**Comment by brettlangdon** [comment]
It depends, there are a few ways to do it.

You can just add your own spans with your own metadata:

```python
from ddtrace import tracer

with tracer.trace("custom.span") as span:
    span.set_tag("metadata.key", "value")
```

Or you can always access the currently active span or the trace root span and add metadata to them that way:

```python
from ddtrace import tracer

span = tracer.current_span()
if span:
    span.set_tag("metadata.key", "value")

root = tracer.current_root_span()
if root:
    root.set_tag("metadata.key", "value")
```

These approaches sort of depend on when you code executes/where you have access to do this.

If the only place where the information to tag with is available is within our instrumentation, there really isn't a good way to extend the instrumentation we have today.

We've had discussions in the past about allowing some kind of "span hook" where you can register a function to be called with a span when it is created, with some additional context (e.g. a boto call with the request and response objects). But we don't have widespread support for this right now.

**Comment by someboredkiddo** [comment]
@brettlangdon - got it, good to know I wasn't crazy. `span hook` sounds cool, but this hasn't come up much and I'd rather you all continue prioritize adding additional contrib tracing. more detailed spans around network calls / sockets would be helpful, we just found a nasty internal dns issue that was exposed by `dd-trace-js` tracing node's `dns` module.  anyway, just my 2 cents on what to prioritize.

**Comment by brettlangdon** [comment]
If there are any contrib tracing you think we are missing, please feel free to open an issue!!

I am very curious to hear your thoughts around more details network/socket tracing, what in Python would be most meaningful for you.

**Pull request merged**